### PR TITLE
Packages update 04-integration

### PR DESCRIPTION
**Pull request by seanfarmar**
Updated packages:
NServiceBus.*
Microsoft.*
Log4Net
Common.Logging
Common.Logging.Core
Autofac
Newtonsoft.Json
NHibernate
Iesi.Collections
Nancy

### CODE REVIEW

File: exercises/04-integration/after/Divergent.Customers.API/Web.config [hunk 1]
```diff
@@ -77,11 +77,10 @@
...
-      <compiler language="vb;vbs;visualbasic;vbscript" extension=".vb" type="Microsoft.CodeDom.Providers.DotNetCompilerPlatform.VBCodeProvider, Microsoft.CodeDom.Providers.DotNetCompilerPlatform, Version=1.0.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" warningLevel="4" compilerOptions="/langversion:14 /nowarn:41008 /define:_MYTYPE=\&quot;Web\&quot; /optionInfer+" />
+      <compiler language="c#;cs;csharp" extension=".cs" type="Microsoft.CodeDom.Providers.DotNetCompilerPlatform.CSharpCodeProvider, Microsoft.CodeDom.Providers.DotNetCompilerPlatform, Version=1.0.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" warningLevel="4" compilerOptions="/langversion:default /nowarn:1659;1699;1701" />
+      <compiler language="vb;vbs;visualbasic;vbscript" extension=".vb" type="Microsoft.CodeDom.Providers.DotNetCompilerPlatform.VBCodeProvider, Microsoft.CodeDom.Providers.DotNetCompilerPlatform, Version=1.0.7.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35" warningLevel="4" compilerOptions="/langversion:default /nowarn:41008 /define:_MYTYPE=\&quot;Web\&quot; /optionInfer+" />
```
Comment (adamralph) [inline exercises/04-integration/after/Divergent.Customers.API/Web.config]: Are the changes to this file (and the similar changes to other config files) required for the package update?

File: exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs [hunk 2]
```diff
@@ -1,23 +0,0 @@
...
-            // NServiceBus will move messages that fail repeatedly to a separate "error" queue. We recommend
-            // that you start with a shared error queue for all your endpoints for easy integration with ServiceControl.
-            endpointConfiguration.SendFailedMessagesTo("error");
-
-            // NServiceBus will store a copy of each successfully process message in a separate "audit" queue. We recommend
-            // that you start with a shared audit queue for all your endpoints for easy integration with ServiceControl.
-            endpointConfiguration.AuditProcessedMessagesTo("audit");
-        }
-    }
-}
```
Comment (adamralph) [inline exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs]: Was this a redundant file? I.e. not included in the project?

File: exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj [hunk 3]
```diff
@@ -211,7 +210,7 @@
         <WebProjectProperties>
           <UseIIS>True</UseIIS>
           <AutoAssignPort>True</AutoAssignPort>
-          <DevelopmentServerPort>20185</DevelopmentServerPort>
+          <DevelopmentServerPort>20186</DevelopmentServerPort>
```
Comment (adamralph) [inline exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj]: Why is this change required? Is it related to the package upgrades?

Comment (seanfarmar) [reply exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj]: No, not related to the package update, but it didn't run properly without this change...

File: exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs (hunk 2 above)
Comment (seanfarmar) [reply exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs]: Not sure, I'll check.

File: exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj (hunk 3 above)
Comment (seanfarmar) [reply exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj]: I'll do the before as well, and squash all to one commit once all is good

File: exercises/04-integration/after/Divergent.Customers.API/Web.config (hunk 1 above)
Comment (seanfarmar) [reply exercises/04-integration/after/Divergent.Customers.API/Web.config]: These were updated when the microsoft.* packages where updated

File: exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs (hunk 2 above)
Comment (seanfarmar) [reply exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs]: yes, it was added by the host package and I accidentally committed it

File: exercises/04-integration/after/Divergent.Customers.API/Web.config (hunk 1 above)
Comment (adamralph) [reply exercises/04-integration/after/Divergent.Customers.API/Web.config]: OK

File: exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs (hunk 2 above)
Comment (adamralph) [reply exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs]: OK, please keep this change as a separate commit.

File: exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj (hunk 3 above)
Comment (adamralph) [reply exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj]: Please leave this change as a separate commit.

Comment (adamralph) [reply exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj]: BTW - I think we need a code change as a result of this. IIRC the host and port of each web API is hardcoded in C#. @Particular/microservices-workshop-maintainers am I correct?

What I can't understand is why I have had no trouble running the solution. What problem did you encounter before changing these ports?

Comment (seanfarmar) [reply exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj]: I had port collisions... so I looked at all the api endpoints and made sure they are correct...

File: exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs (hunk 2 above)
Comment (seanfarmar) [reply exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs]: I introduced this file while I was doing the updates AFAIK, so I squashed it...

Comment (adamralph) [reply exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs]: This file has to be present in master right now, otherwise it wouldn't show as a removal in this PR.

File: exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj (hunk 3 above)
Comment (adamralph) [reply exercises/04-integration/after/Divergent.Customers.API/Divergent.Customers.API.csproj]: I don't have this problem. Everything runs OK when I run the solution as-is. I don't think I saw any workshop attendees having this problem either. @Particular/microservices-workshop-maintainers does anyone know why @seanfarmar is having this problem?

Regardless, would it be best to remove it from this PR and deal with this separately? If we can keep this PR isolated to package updates only then we can get it merged in quicker.

File: exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs (hunk 2 above)
Comment (seanfarmar) [reply exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs]: Yes you are right! It is on master...

What do you think we should do?

**Comment by seanfarmar** [comment]
Reverted the ports change commit, so now we only have the packages updates for `before` and `after` solutions

it also includes the deletion of the redundant `EndpointConfig.cs` file

File: exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs (hunk 2 above)
Comment (adamralph) [reply exercises/04-integration/after/Divergent.Customers/EndpointConfig.cs]: As discussed, I'm OK to mix this minor change with the package update.

**Comment by seanfarmar** [comment]
@adamralph I pushed a version that removed the endpointconfig.cs file deletion... But i forced pushed it, so your changes have been overwritten... can you review again and re-introduce them please?

**Comment by seanfarmar** [comment]
@adamralph LGTM :-)
